napari/layers/shapes/_shapes_mouse_bindings.py

```diff
@@ -681,6 +682,149 @@ def _set_drag_start(
     return coord
 
 
+def _move_selected_or_add_rectangle_ellipse_line(
+    layer: Shapes, coordinates: tuple[float, ...], vertex: int
+) -> None:
```

Comment:
why moving this code to another function if it anyways stays is one big block and the outer function becomes just a thin wrapper?

Either way, this name is super confusing, I'm not sure what this function is actually supposed to do anymore...

Reply:
I moved here single if branch. So function from where I get it becomes more readable. 

I do not have a good idea for function name. 

The moving shape and creating `Rectange`, `Ellipse` or `Line` should be splitted.

Reply:
> The moving shape and creating Rectange, Ellipse or Line should be splitted.

100% agreed 😂 do you think that you could achieve this as part of this PR @Czaki? Looking at the calling function, and since we are dropping 3.9, this is probably a really good place to use the Python 3.10+ `match` statement. ([old discussion in Zulip](https://napari.zulipchat.com/#narrow/channel/348229-randoms/topic/match.2Fcase.20in.20Python.203.2E10.2B/near/407454349))

Reply:
Ok, actually, thinking about this: CI won't pass until we switch to 3.10+, and at that point, pre-commit makes a ton of unrelated changes. So maybe we merge this ~as is, then we postpone the match version until after the migration?